test/step-function.test.ts

```diff
@@ -0,0 +1,5198 @@
+        OutputPath: "$.null",
+      },
+    },
+  };
+  expect(definition).toEqual(expected);
+});
+
+test("for i in items, items[i]", () => {
+  const { stack } = initStepFunctionApp();
+  const definition = new ExpressStepFunction(stack, "fn", (items: string[]) => {
+    for (const i in items) {
+      // @ts-ignore
+      const a = items[i];
+    }
+  }).definition;
+
+  expect(definition).toEqual({
+    StartAt: "for(i in items)",
+    States: {
+      "for(i in items)": {
+        ItemsPath: "$.items",
+        Iterator: {
+          StartAt: "a = items[i]",
+          States: {
+            "a = items[i]": {
+              End: true,
+              OutputPath: "$.result",
+              Parameters: {
+                "result.$": "$.0_i",
+              },
+              ResultPath: "$.a",
+              Type: "Pass",
+            },
+          },
+        },
+        MaxConcurrency: 1,
+        Next: "return null",
+        Parameters: {
+          // special prefixed value so that we can index the
+          "0_i.$": "$$.Map.Item.Value",
+          "i.$": "$$.Map.Item.Index",
+        },
+        ResultPath: null,
+        Type: "Map",
+      },
+      "return null": {
+        End: true,
+        OutputPath: "$.null",
+        Parameters: {
+          null: null,
+        },
+        Type: "Pass",
+      },
+    },
+  });
+});
+
+test("return a single Lambda Function call", () => {
+  const { stack, getPerson } = initStepFunctionApp();
+  const definition = new ExpressStepFunction(
+    stack,
+    "fn",
+    (id: string): Person | undefined => {
+      return getPerson({ id });
+    }
+  ).definition;
+
+  expect(definition).toEqual({
+    StartAt: "return getPerson({id: id})",
+    States: {
+      "return getPerson({id: id})": {
+        Type: "Task",
+        Resource: "arn:aws:states:::lambda:invoke",
+        End: true,
+        ResultPath: "$",
+        Parameters: {
+          FunctionName: getPerson.resource.functionName,
+          Payload: {
+            "id.$": "$.id",
+          },
+        },
+      },
+    },
+  });
+});
+
+test("call Lambda Function, store as variable, return variable", () => {
+  const { stack, getPerson } = initStepFunctionApp();
+  const definition = new ExpressStepFunction(
+    stack,
+    "fn",
+    (id: string): Person | undefined => {
+      const person = getPerson({ id });
+      return person;
+    }
+  ).definition;
+
+  expect(definition).toEqual({
+    StartAt: "person = getPerson({id: id})",
+    States: {
+      "person = getPerson({id: id})": {
+        Type: "Task",
+        Resource: "arn:aws:states:::lambda:invoke",
+        ResultPath: "$.person",
+        Parameters: {
+          FunctionName: getPerson.resource.functionName,
+          Payload: {
+            "id.$": "$.id",
+          },
+        },
+        Next: "return person",
+      },
+      "return person": {
+        Type: "Pass",
+        OutputPath: "$.person",
+        End: true,
+      },
+    },
+  });
+});
```

Comment:
What do you expect lambda to respond with? This integration returns the entire lambda http response.

If you want the payload only, you can give the lambda arn only and not the `:invoke`

https://github.com/aws/aws-cdk/blob/v1.150.0/packages/@aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke.ts#L146-L151

Reply:
Good catch - I don't think I've tested this live.

Reply:
Also wait/token use case would be interesting to model.

Reply:
https://github.com/sam-goodwin/functionless/issues/79